include/openthread/message.h

```diff
@@ -69,6 +69,19 @@ typedef enum otMessagePriority
     OT_MESSAGE_PRIORITY_HIGH   = 2, ///< High priority level.
 } otMessagePriority;
 
+/**
+ * Defines the OpenThread message origins.
+ *
+ */
+typedef enum otMessageOrigin
+{
+    OT_MESSAGE_ORIGIN_FROM_THREAD_NETIF           = 0, ///< Message originates from Thread Netif.
```

Comment:
How about we remove `_FROM_` and use `OT_MESSAGE_ORIGIN_THREAD_NETIF`, `OT_MESSAGE_ORIGIN_HOST_DISALLOW_LOOPBACK`, ect.

We used `kFromThreadNetif` in `Ip6` class `enum` since we did not include the term `Origin` in the constants and `kFrom` helped make it more clear and readable but since we now have `MESSAGE_ORIGIN_` prefix  we don't really need the `FROM_` anymore.